app/src/main/java/com/marvel/marvel/customview/ComicsHolder.kt

```diff
@@ -0,0 +1,48 @@
+package com.marvel.marvel.customview
+
+import android.support.v7.widget.RecyclerView
+import android.view.View
+import com.marvel.marvel.application.ImageLoader
+import com.marvel.marvel.viewmodel.ComicsViewModel
+import kotlinx.android.synthetic.main.comics_item.view.*
+
+class ComicsHolder(
+    itemView: View,
+    private val imageLoader: ImageLoader
+) : RecyclerView.ViewHolder(itemView) {
+
+    companion object {
+        private const val CLICK_DELAY = 500
+    }
+
+    private var model: ComicsViewModel? = null
+    private var callback: MarvelRecyclerView.Callback? = null
+    private var time: Long? = null
+
+    fun bind(model: ComicsViewModel, callback: MarvelRecyclerView.Callback?) {
+        this.model = model
+        this.callback = callback
+        imageLoader.loadImage(itemView.image, model.imageUrl,
+            object : ImageLoader.SimpleCallback() {
+                override fun onSuccess() {
+                    itemView.setOnClickListener {
+                        onItemClicked()
+                    }
+                }
+            })
+    }
+
+    private fun onItemClicked() {
+        val lastClickTime = time ?: 0
+        if (System.currentTimeMillis() - lastClickTime > CLICK_DELAY) {
+            callback?.onItemClicked(model, itemView.image)
+            time = System.currentTimeMillis()
+        }
```

Comment:
I suggest you look into rx [debounce](http://rxmarbles.com/#debounce) and Jake Wharton's [RxBindings](https://github.com/JakeWharton/RxBinding).

```
callback?.let { cb ->
subscription.add(itemView
    .clicks() // view clicks as an observable RX stream!
    .debounce(500, TimeUnit.Ms)
    .subscribe {
         cb.onItemClicked(model, itemView.image)
    }
})
```
Sub on `bind` and unsub on `unbind`
And I wouldn't set this up on Picasso call back, prob just check the presence of an image and clear images out on unbind.

Reply:
Good idea checking if bitmap is there rather that messing with callbacks...